### PR TITLE
ci(framework:skip) Add job to update translation text

### PR DESCRIPTION
**Pull request by charlesbvll**
### Description

For our translation tool (Weblate) to use the latest doc changes as a translation source, we need to update the `.po` files under our `doc/locales` folders. This can be done manually, but it would be nice to have an automatic way of doing it. We could run a workflow on doc PRs but that would mean making automatic commits to already opened PR, which doesn't necessarily seem like a good idea.

### Related issues/PRs

N/A

### Explanation

We schedule a CI job that runs every day at midnight and that can also be triggered manually in order to update the translation source text. This proposal has been tested on a private fork of the repo.

### Checklist

- [x] Implement proposed change
- [x] Write tests
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### Any other comments?

N/A

### CODE REVIEW

File: .github/workflows/update_trans_docs_text.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          done
+
+      - name: Commit changes
+        id: commit_changes
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add doc/locales
+          git commit -m "Update text and language files" || echo "::set-output name=changes::false"
+          echo "::set-output name=changes::true"
```
[the remark this answers is not in the document]
Comment (charlesbvll) [reply .github/workflows/update_trans_docs_text.yml:52]: Oh right ^^ I think we can remove this anyways, as the condition on the diff is stronger

File: .github/workflows/update_trans_docs_text.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        continue-on-error: true
+
+      - name: Calculate diff  # Even without doc changes the update-lang command will generate 228 additions and 60 deletions, so we only want to open a PR when there is more
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply .github/workflows/update_trans_docs_text.yml:53]: Sounds good! If we have to readjust it too often, we can look for another solution.

**Review by Robert-Steiner — APPROVED** [review]
Happy to see that this is now automated!

**Review by tanertopal — APPROVED** [review]
Lgtm!

**Pull request merged**